core/src/main/java/org/jdbi/v3/BasicHandle.java

```diff
@@ -242,30 +244,32 @@ public boolean isInTransaction()
     @Override
     public Update createStatement(String sql)
     {
+        Foreman updateForeman = foreman.createChild();
         return new Update(this,
                           statementLocator,
                           statementRewriter,
                           statementBuilder,
                           sql,
-                          new ConcreteStatementContext(globalStatementAttributes, MappingRegistry.copyOf(mappingRegistry)),
+                          new ConcreteStatementContext(globalStatementAttributes, MappingRegistry.copyOf(mappingRegistry), updateForeman),
                           timingCollector,
-                          foreman,
+                          updateForeman,
                           collectorFactoryRegistry);
```

Comment:
I think so, otherwise registering on the `Update` will cascade up to the `Handle`.